WooCommerce/Classes/POS/Presentation/Barcode Scanning/BarcodeScannerContainer.swift

```diff
@@ -39,15 +39,23 @@ struct BarcodeScannerContainerRepresentable: UIViewControllerRepresentable {
     let configuration: HIDBarcodeParserConfiguration
     let onScan: (Result<String, Error>) -> Void
 
-    func makeUIViewController(context: Context) -> BarcodeScannerHostingController {
-        let controller = BarcodeScannerHostingController(
-            configuration: configuration,
-            onScan: onScan
-        )
-        return controller
+    func makeUIViewController(context: Context) -> UIViewController {
+        let featureFlagService = ServiceLocator.featureFlagService
+
+        if featureFlagService.isFeatureFlagEnabled(.pointOfSaleBarcodeScanningi2) {
+            return GameControllerBarcodeScannerHostingController(
+                configuration: configuration,
+                onScan: onScan
+            )
+        } else {
+            return BarcodeScannerHostingController(
+                configuration: configuration,
+                onScan: onScan
+            )
+        }
```

Comment:
```suggestion
        return GameControllerBarcodeScannerHostingController(
            configuration: configuration,
            onScan: onScan
        )
```



I don't actually think this needs to be feature flagged, we can just ship it – we're changing the app to match Android i1 after all. 

It is helpful for testing the PR though. If you want to keep the flag, perhaps rename it to something other than i2, so we can ship this before the rest of i2.

Reply:
Yes, it was helpful for testing. I didn't want to remove it right away. However, we could simply replace one with another without any feature flagging.